### PR TITLE
Activity.Baggage values are now be Url decoded in HostingApplicationD…

### PR DESCRIPTION
**Pull request by vaughanr**
Summary of the changes (Less than 80 chars)
 - This change url decodes the value of the activity baggage since it is url encoded when sent to the host.

Addresses #18600

### CODE REVIEW

**Review by JunTaoLuo — APPROVED** [review]

**Comment by JunTaoLuo** [comment]
Looks good, I'm going to re-run the tests before merging though I believe the hang is completely unrelated.

**Comment by vaughanr** [comment]
Hi John, I first rebased the commit to the latest master but that didn't fix the build. I think the issue was potentially that the urldecode method was in a system.net lib that might not be installed in some environments. I switched to the System.Web.HttpUtility version and that seems to have fixed it.

**Comment by Tratcher** [comment]
> I first rebased the commit to the latest master but that didn't fix the build. I think the issue was potentially that the urldecode method was in a system.net lib that might not be installed in some environments. I switched to the System.Web.HttpUtility version and that seems to have fixed it.

? Neither commit added any assembly references so missing dlls shouldn't be a problem. What was the original error?

**Comment by vaughanr** [comment]
To be honest I don't know why the previous one always failed. I assumed it might be that some environments don't ship with .net framework installed and so I would have had to explicitly added the system.net dependency. I created a temp PR to compare the previous commit to this one.

I don't really understand what the error is because the errors I see seem unrelated to code changes. This time it failed in building the Windows x64/x86 task.

https://dev.azure.com/dnceng/public/_build/results?buildId=527409&view=logs&jobId=1b89928a-2219-5ef9-602f-f95beb3da4dc&j=1b89928a-2219-5ef9-602f-f95beb3da4dc

**Comment by Tratcher** [comment]
Our tests are subject to sporadic flakiness due to test or infrastructure issues. Sorry that disrupted your PR. If you see a failure that does not make sense just let us know.

**Comment by vaughanr** [comment]
@Tratcher it's not a problem, the fact that it failed without explanation most probably merits your question. I'd rather not break everyone's build with my first commit so am happy to have another pair of eyes!

@JunTaoLuo are you still happy with this or would you prefer me to change it back to the system.net.webutility method?

**Comment by JunTaoLuo** [comment]
As far as I can tell System.Web has a bit more extra logic to handle %uXXXX in addition to %XX. Otherwise the behaviour seems pretty much the same. I think either API would be sufficient so I'm just going to merge this as is.

**Comment by vaughanr** [comment]
Thank you!

**Comment by Vfialkin** [comment]
Thank you for fixing this.
One note on a subject. It was a bit surprising to find out that the same fix was applied and merged to 3.1.10 - https://github.com/dotnet/aspnetcore/pull/26398/files on behalf of another ticket.
Hope that will help someone who finds that breaking change in 3.1 😉

**Pull request merged**
